pkg/apiserver/resthandler.go

```diff
@@ -269,8 +269,14 @@ func CreateResource(r rest.Creater, scope RequestScope, typer runtime.ObjectType
 			errorJSON(err, scope.Codec, w)
 			return
 		}
+		// Get name, ok if empty
+		_, name, _ := scope.Namer.Name(req)
+
 		ctx := scope.ContextFunc(req)
 		ctx = api.WithNamespace(ctx, namespace)
+		if len(name) > 0 {
+			ctx = api.WithValue(ctx, "name", name)
```

Comment:
It's the Storage objects' responsibility to make that determination, by passing it the name and the object.  Then the storage object can do whatever it wants, including returning an error or silently eating it.

----- Original Message -----
> >  		ctx := scope.ContextFunc(req)
> >  		ctx = api.WithNamespace(ctx, namespace)
> > +		if len(name) > 0 {
> > +			ctx = api.WithValue(ctx, "name", name)
> 
> If we set the name on the object we may be hiding an error condition. You may
> be posting to a particular path and post an object with a different name. I
> would think you'd want the server to tell you that you have a mismatch
> instead of just taking the name from the path.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7718/files#r29630615